compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -421,36 +421,19 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
             rewrite(field, recordTypeNode.typeDefEnv);
         }
         recordTypeNode.restFieldType = rewrite(recordTypeNode.restFieldType, env);
-        // In the current implementation, closures generated for default values in inclusions defined in a
-        // separate module are unidentifiable.
-        // Due to that, if the inclusions are in different modules, we generate closures again.
-        // Will be fixed  with #41949 issue.
-        generateClosuresForDefaultValuesInTypeInclusionsFromDifferentModule(recordTypeNode);
+        generateClosuresForNonOverriddenFields(recordTypeNode);
         result = recordTypeNode;
     }
 
-    private List<String> getFieldNames(List<BLangSimpleVariable> fields) {
-        List<String> fieldNames = new ArrayList<>();
-        for (BLangSimpleVariable field : fields) {
-            fieldNames.add(field.name.getValue());
-        }
-        return fieldNames;
-    }
-
-    private void generateClosuresForDefaultValuesInTypeInclusionsFromDifferentModule(
-            BLangRecordTypeNode recordTypeNode) {
+    private void generateClosuresForNonOverriddenFields(BLangRecordTypeNode recordTypeNode) {
         if (recordTypeNode.typeRefs.isEmpty()) {
             return;
         }
         List<String> fieldNames = getFieldNames(recordTypeNode.fields);
         BTypeSymbol typeSymbol = recordTypeNode.getBType().tsymbol;
         String typeName = recordTypeNode.symbol.name.value;
-        PackageID packageID = typeSymbol.pkgID;
         for (BLangType type : recordTypeNode.typeRefs) {
             BType bType = type.getBType();
-            if (packageID.equals(bType.tsymbol.pkgID)) {
-                continue;
-            }
```

Comment:
Given that we're removing this, I think we can improve `generateClosureForDefaultValues` to create new closures only if the original closure is defined in a separate module?

Reply:
I guess we can do this separately though.

Reply:
To clarify in the fallowing code,
```
type OuterXBase record {
    Inner inner = {foo: 10};
};

type OuterX record {|
    *OuterXBase; // <- default value must come from the inclusion
|};
```

In current implementation we create a lambda for creating an `Inner` value as a default closure for `OuterXBase` (let's call this `f1`), and for `OuterX` we create a lambda (let's call this `f2`) that simply invoke `f1`.  Are you suggesting in both cases they should have a single lambda (`f1`)

Reply:
Added a comment to  #41949